src/openrct2/world/TileElement.h

```diff
@@ -33,6 +33,36 @@ class FootpathRailingsObject;
 enum class RideColourScheme : uint8_t;
 using track_type_t = uint16_t;
 
+struct CarTrackLocation : public CoordsXYZD
```

Comment:
This doesn't have anything to do with tile elements, so `TileElement.h` doesn't feel appropriate place for it. Considering it's only really used by `ScVehicle.cpp` and `Duktape.hpp`, I think it should be moved to a more local header.

@Broxzier Do you have any thoughts on this?

Reply:
true. I really just shoved it into the first place that made it compile without warnings.

Reply:
done. moved to the buttom of Duktape.hpp.

Reply:
woops turns out I don't know how to do forward declerations for structs that inherit stuff in c++ 😜 
but I also struggle with adding new files to the project 😂. usually for c++ projects you have a list of files somewhere you need to append... Couldn't find it.

Reply:
ugh just moved the entire struct definition to the _top_ of that file. Open for improvement.